overlord/ifacestate/ifacestate_test.go

```diff
@@ -10696,3 +10696,134 @@ func (s *interfaceManagerSuite) TestOnSnapLinkageChanged(c *C) {
 		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{&info.SideInfo}),
 	})
 }
+
+func (s *interfaceManagerSuite) TestSetupProfilesAffectedSnapRegenUsesMostRecentRevision(c *C) {
+	// this test checks a bug fix where a consumer and producer snap (content interface)
+	// were refreshed together and sometimes the producing snap lost access to its own
+	// file. The issue was that, if the slot (producer) was refreshed first, then,
+	// when the plugging snap was refreshed it would regenerate the slots profile
+	// but with the previous revision, not the revision that the current change was
+	// updating to.
+	s.mockIfaces(&ifacetest.TestInterface{InterfaceName: "test"})
+
+	const componentYaml = `
+component: producer2+comp1
+type: standard
+version: 1.0
+`
+	compProducerYaml := producer2Yaml +
+		`components:
+  comp1:
+    type: standard
+`
+
+	mgr := s.manager(c)
+	repo := mgr.Repository()
+
+	consumer := s.mockSnap(c, consumer2Yaml)
+	producer := s.mockSnap(c, compProducerYaml)
+
+	s.state.Lock()
+	var snapst snapstate.SnapState
+	c.Assert(snapstate.Get(s.state, "producer2", &snapst), IsNil)
+
+	compInfo := snaptest.MockComponent(c, componentYaml, producer, snap.ComponentSideInfo{
+		Revision: snap.R(1),
+	})
+
+	err := snapst.Sequence.AddComponentForRevision(snap.R(1), &sequence.ComponentState{
+		SideInfo: &compInfo.ComponentSideInfo,
+		CompType: snap.StandardComponent,
+	})
+	c.Assert(err, IsNil)
+	snapstate.Set(s.state, "producer2", &snapst)
+
+	for _, info := range []*snap.Info{consumer, producer} {
+		appSet, err := interfaces.NewSnapAppSet(info, nil)
+		c.Assert(err, IsNil)
+
+		err = repo.AddAppSet(appSet)
+		c.Assert(err, IsNil)
+	}
+
+	connRef := &interfaces.ConnRef{PlugRef: interfaces.PlugRef{
+		Snap: "consumer2",
+		Name: "plug",
+	}, SlotRef: interfaces.SlotRef{
+		Snap: "producer2",
+		Name: "slot",
+	}}
+	repo.Connect(connRef, nil, nil, nil, nil, nil)
```

Comment:
Can we check the return value (error) here please?

Reply:
Good point, it's done now. thanks